### PR TITLE
TSPS-174  renaming non cosmetic things from tsps to teaspoons

### PR DESCRIPTION
**Pull request by jsotobroad**
### Description 

We decided to rename out internal short name from tsps to teaspoons.  This is making the changes in our repo first so we know what downstream things we should look into changing and try to merge the changes in logical order

working workflow with the consolidated db changesets - https://bvdp-saturn-dev.appspot.com/#workspaces/teaspoons-imputation-dev/teaspoons_imputation_dev_control_workspace_20240726/job_history/89bce976-a937-4829-bbe7-ad7d4e5dece8

### Jira Ticket
https://broadworkbench.atlassian.net/browse/TSPS-174

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -138,8 +138,8 @@ jobs:
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
         with:
           status: failure
-          channel: "#terra-tsps-alerts"
-          username: "TSPS push to main branch"
+          channel: "#terra-teaspoons-alerts"
```
Comment (jsotobroad) [inline .github/workflows/build-and-test.yml:141]: how hard is it to change slack channel names?

File: README.md [hunk 2]
```diff
@@ -82,7 +82,7 @@ SonarQube and want to debug the problem locally, you need to get the sonar token
 before running the gradle task.
 
 ```shell
-export SONAR_TOKEN=$(vault read -field=sonar_token secret/secops/ci/sonarcloud/tsps)
+export SONAR_TOKEN=$(vault read -field=sonar_token secret/secops/ci/sonarcloud/teaspoons)
```
Comment (jsotobroad) [inline README.md:85]: do we want to change our vault path?

File: README.md [hunk 3]
```diff
@@ -109,13 +109,13 @@ Upon merging to main, the dev environment will be automatically deployed via the
...
+You can check the status of the deployment in [Beehive](https://beehive.dsp-devops.broadinstitute.org/apps/teaspoons) and in 
+[ArgoCD](https://ap-argocd.dsp-devops.broadinstitute.org/applications/ap-argocd/teaspoons-dev).
 
 For more information about deployment to dev, check out DevOps' [excellent documentation](https://docs.google.com/document/d/1lkUkN2KOpHKWufaqw_RIE7EN3vN4G2xMnYBU83gi8VA/).
 
 ### Tracing
 
 We use [OpenTelemetry](https://opentelemetry.io/) for tracing, so that every request has a tracing span that can 
-be viewed in [Google Cloud Trace](https://cloud.google.com/trace). (This is not yet fully set up here - to be done in TSPS-107). 
+be viewed in [Google Cloud Trace](https://cloud.google.com/trace). (This is not yet fully set up here - to be done in Teaspoons-107).
```
Comment (jsotobroad) [inline README.md]: can we change the shortname of our jira board/tickets easily and what do we rename it to if it is?

File: service/src/main/resources/application.yml [hunk 4]
```diff
@@ -9,7 +9,7 @@ env:
       pass: ${DATABASE_USER_PASSWORD:dbpwd}
       user: ${DATABASE_USER:dbuser}
     stairway:
-      name: ${STAIRWAY_DATABASE_NAME:tsps_stairway_db}
+      name: ${STAIRWAY_DATABASE_NAME:teaspoons_stairway_db}
```
Comment (jsotobroad) [inline service/src/main/resources/application.yml:12]: def need to make sure these database values are changed upstream

File: README.md [hunk 5]
```diff
@@ -109,13 +109,13 @@ Upon merging to main, the dev environment will be automatically deployed via the
...
+You can check the status of the deployment in [Beehive](https://beehive.dsp-devops.broadinstitute.org/apps/teaspoons) and in 
+[ArgoCD](https://ap-argocd.dsp-devops.broadinstitute.org/applications/ap-argocd/teaspoons-dev).
 
 For more information about deployment to dev, check out DevOps' [excellent documentation](https://docs.google.com/document/d/1lkUkN2KOpHKWufaqw_RIE7EN3vN4G2xMnYBU83gi8VA/).
 
 ### Tracing
 
 We use [OpenTelemetry](https://opentelemetry.io/) for tracing, so that every request has a tracing span that can 
-be viewed in [Google Cloud Trace](https://cloud.google.com/trace). (This is not yet fully set up here - to be done in TSPS-107). 
+be viewed in [Google Cloud Trace](https://cloud.google.com/trace). (This is not yet fully set up here - to be done in TEASPOONS-107).
```
Comment (mmorgantaylor) [inline README.md:120]: did we decide on a new Jira handle yet?

File: service/src/main/java/bio/terra/pipelines/dependencies/cbas/CbasService.java [hunk 6]
```diff
@@ -43,11 +43,11 @@ public MethodListResponse getAllMethods(String cbasBaseUri, String accessToken)
    * .methodDescription("method description") //
    * .methodSource(PostMethodRequest.MethodSourceEnum.GITHUB) // .methodUrl(pipeline.getWdlUrl()) //
    * .methodVersion("1.0"); // logger.info( // "this is creating a new method in cbas: {}", //
-   * cbasService.createMethod( // cbasUri, samService.getTspsServiceAccountToken(),
+   * cbasService.createMethod( // cbasUri, samService.getTeasponnsServiceAccountToken(),
```
[the remark this answers is not in the document]
Comment (jsotobroad) [reply service/src/main/java/bio/terra/pipelines/dependencies/cbas/CbasService.java:46]: its so hard to spell it sometimes!!!

File: service/src/main/java/bio/terra/pipelines/dependencies/sam/SamService.java [hunk 7]
```diff
@@ -65,15 +65,15 @@ public ApiSystemStatusSystems checkHealthApiSystemStatus() {
...
+  public String getTeaspoonsServiceAccountToken() {
     try {
       GoogleCredentials creds =
           GoogleCredentials.getApplicationDefault().createScoped(SAM_OAUTH_SCOPES);
       creds.refreshIfExpired();
       return creds.getAccessToken().getTokenValue();
     } catch (IOException e) {
       throw new InternalServerErrorException(
-          "Internal server error retrieving TSPS credentials", e);
+          "Internal server error retrieving Teaspoons credentials", e);
```
[the remark this answers is not in the document]
Comment (jsotobroad) [reply service/src/main/java/bio/terra/pipelines/dependencies/sam/SamService.java:76]: oh thats a good point, should we say `Terra Scientific Pipeline Services instead`?

File: README.md (hunk 5 above)
Comment (jsotobroad) [reply README.md:120]: i dont think we have.  I thougth i put a comment about that somewhere but i dont see it =/

File: service/src/main/java/bio/terra/pipelines/App.java [hunk 8]
```diff
@@ -26,7 +26,7 @@
       "bio.terra.common.retry.transaction",
       // Stairway initialization and status
       "bio.terra.common.stairway",
-      // Scan all TSPS service packages
+      // Scan all Teaspoons service packages
```
[the remark this answers is not in the document]
Comment (jsotobroad) [reply service/src/main/java/bio/terra/pipelines/App.java:29]: i actually like pipelines.  Similar to like workspace manager, their package name is workspace

File: service/src/main/java/bio/terra/pipelines/dependencies/stairway/JobService.java [hunk 9]
```diff
@@ -155,7 +155,8 @@ public <T> JobResultOrException<T> retrieveJobResult(
 
       switch (flightState.getFlightStatus()) {
         case FATAL:
-          logAlert("TSPS Stairway flight {} encountered dismal failure", flightState.getFlightId());
+          logAlert(
+              "Teaspoons Stairway flight {} encountered dismal failure", flightState.getFlightId());
```
[the remark this answers is not in the document]
Comment (jsotobroad) [reply service/src/main/java/bio/terra/pipelines/dependencies/stairway/JobService.java:159]: removed

File: service/src/main/resources/db/changesets/20241106-base-data-insertion.yaml [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+                  value: Array Imputation
+              - column:
+                  name: description
+                  value: Phase and impute genotypes using Beagle 5.4 with the AoU/AnVIL reference panel of 515,579 samples.
+              - column:
+                  name: pipeline_type
+                  value: imputation
+              - column:
+                  name: wdl_url
+                  value: https://github.com/broadinstitute/warp/blob/TSPS-183_mma_beagle_imputation_hg38/pipelines/broad/arrays/imputation_beagle/ImputationBeagle.wdl
```
Comment (jsotobroad) [inline service/src/main/resources/db/changesets/20241106-base-data-insertion.yaml:28]: wasnt sure if this was correct but i think this is the latest value we have

File: .github/workflows/run-e2e-tests.yaml [hunk 11]
```diff
@@ -89,7 +89,7 @@ jobs:
     permissions:
       contents: read
       id-token: write
-    uses: broadinstitute/dsp-reusable-workflows/.github/workflows/run_tsps_e2e_tests.yaml@main
+    uses: broadinstitute/dsp-reusable-workflows/.github/workflows/run_teaspoons_e2e_tests.yaml@main
```
Comment (mmorgantaylor) [inline .github/workflows/run-e2e-tests.yaml:92]: we'll need to open a PR in dsp-reusable-workflows to update the names of all the relevant files - they all say "tsps" currently, e.g. https://github.com/broadinstitute/dsp-reusable-workflows/blob/main/.github/workflows/run_tsps_e2e_tests.yaml

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (mmorgantaylor) [reply .github/workflows/build-and-test.yml:139]: when should we make this change? right before merging this PR?

File: README.md [hunk 12]
```diff
@@ -106,7 +106,7 @@ does all the setup for you. Clone that repo and make sure you're either on Broad
 to the VPN. Then run the following command:
 
 ```shell
-./db/psql-connect.sh dev tsps
+./db/psql-connect.sh dev teaspoons
```
Comment (mmorgantaylor) [inline README.md:109]: does this work already? any changes we need to make in that script?

File: README.md [hunk 13]
```diff
@@ -129,7 +129,7 @@ See [this DSP blog post](https://broadworkbench.atlassian.net/wiki/x/AoGlrg) for
 ### Running the end-to-end tests
 
 The end-to-end test is specified in `.github/workflows/run-e2e-tests.yaml`. It calls [the test script defined 
-in the dsp-reusable-workflows repo](https://github.com/broadinstitute/dsp-reusable-workflows/blob/main/e2e-test/tsps_e2e_test.py).
+in the dsp-reusable-workflows repo](https://github.com/broadinstitute/dsp-reusable-workflows/blob/main/e2e-test/teaspoons_e2e_test.py).
```
Comment (mmorgantaylor) [inline README.md:132]: current path is `https://github.com/broadinstitute/dsp-reusable-workflows/blob/main/e2e-test/tsps_gcp_e2e_test.py`; we should change it in that repo and here to `https://github.com/broadinstitute/dsp-reusable-workflows/blob/main/e2e-test/teaspoons_gcp_e2e_test.py`

File: service/src/main/resources/db/changelog.xml [hunk 14]
```diff
@@ -3,36 +3,10 @@
...
-  <include file="changesets/20240830_add_workspace_google_project_rename_storage_container_field.yaml" relativeToChangelogFile="true"/>
-  <include file="changesets/20240903_rename_workspace_project.yaml" relativeToChangelogFile="true"/>
-  <include file="changesets/20240906_add_wdl_version_to_pipeline_run.yaml" relativeToChangelogFile="true"/>
-  <include file="changesets/20240911_remove_isSuccess.yaml" relativeToChangelogFile="true"/>
-  <include file="changesets/20240923_update_input_defs_bucket_paths.yaml" relativeToChangelogFile="true"/>
-  <include file="changesets/20240925_update_ref_panel_prefix_input.yaml" relativeToChangelogFile="true"/>
-  <include file="changesets/20241030_add_pipeline_quotas.yaml" relativeToChangelogFile="true"/>
-  <include file="changesets/20241103_add_user_quotas.yaml" relativeToChangelogFile="true"/>
-  <include file="changesets/20241106_array_imputation_rename.yaml" relativeToChangelogFile="true"/>
+  <include file="changesets/20241106-base-tables.yaml" relativeToChangelogFile="true"/>
```
Comment (mmorgantaylor) [inline service/src/main/resources/db/changelog.xml:6]: 😍

File: service/src/main/resources/db/changesets/20241106-base-data-insertion.yaml (hunk 10 above)
Comment (mmorgantaylor) [reply service/src/main/resources/db/changesets/20241106-base-data-insertion.yaml:28]: i think that's right too (it matches what we currently have)

File: service/src/test/java/bio/terra/pipelines/common/utils/StairwaySetPipelineRunStatusHookTest.java [hunk 15]
```diff
@@ -7,6 +7,7 @@
 
 import bio.terra.pipelines.db.entities.PipelineRun;
 import bio.terra.pipelines.db.repositories.PipelineRunsRepository;
+import bio.terra.pipelines.db.repositories.PipelinesRepository;
```
Comment (mmorgantaylor) [inline service/src/test/java/bio/terra/pipelines/common/utils/StairwaySetPipelineRunStatusHookTest.java:10]: was this just missing by accident before?

**Review by mmorgantaylor — COMMENTED** [review]
looking great, a couple comments/questions. is the next step to walk through this with Chelsea?

File: service/src/test/java/bio/terra/pipelines/common/utils/StairwaySetPipelineRunStatusHookTest.java (hunk 15 above)
Comment (jsotobroad) [reply service/src/test/java/bio/terra/pipelines/common/utils/StairwaySetPipelineRunStatusHookTest.java:10]: huh i have no idea why that was added, i removed it and tests still seem to be passing.  I prob messed up something while i was refactoring stuff 🤷

**Comment by jsotobroad** [comment]
> looking great, a couple comments/questions. is the next step to walk through this with Chelsea?

yea I think the next step would be to chat with chelsea for the vault stuff and then co-ordinate the new CI with this codebase with a clean DB.  For the slack and e2e test renaming stuff i figure we can knock that out as we merge this pr in

File: README.md (hunk 12 above)
Comment (jsotobroad) [reply README.md:109]: it does not, we'd have to make all the related tsps -> teaspoon changes affter this pr gets merged

File: service/src/main/resources/application.yml [hunk 16]
```diff
@@ -25,7 +25,7 @@ env:
     authorityEndpoint: ${OIDC_AUTHORITY_ENDPOINT:https://terradevb2c.b2clogin.com/terradevb2c.onmicrosoft.com/oauth2/v2.0/authorize?p=b2c_1a_signup_signin_dev}
     tokenEndpoint: ${OIDC_TOKEN_ENDPOINT:https://terradevb2c.b2clogin.com/terradevb2c.onmicrosoft.com/oauth2/v2.0/token?p=b2c_1a_signup_signin_dev}
   ingress:
-    domainName: ${TSPS_INGRESS_DOMAIN_NAME:localhost:8080}
+    domainName: ${TEAPSOONS_INGRESS_DOMAIN_NAME:localhost:8080}
```
Comment (mmorgantaylor) [inline service/src/main/resources/application.yml:28]: is this env var all set up (probably a question for @choover-broad)

**Pull request merged**